pkg/kubectl/rolling_updater.go

```diff
@@ -411,13 +417,14 @@ func (r *RollingUpdater) readyPods(oldRc, newRc *api.ReplicationController) (int
 			return 0, 0, err
 		}
 		for _, pod := range pods.Items {
-			if api.IsPodReady(&pod) {
-				switch controller.Name {
-				case oldRc.Name:
-					oldReady++
-				case newRc.Name:
-					newReady++
-				}
+			if r.nowFn != nil && !deployment.IsPodAvailable(&pod, minReadySeconds, r.nowFn().Time) {
```

Comment:
@kargakis the nowFn is defaulted in initializer:

```
	updater.nowFn = func() unversioned.Time { return unversioned.Now() }
```

but yes, in case you don't use initializer, we should default there (tests)